src/SourceLocator/Type/Composer/Factory/MakeLocatorForComposerJsonAndInstalledJson.php

```diff
@@ -38,19 +38,22 @@ public function __invoke(string $installationPath, Locator $astLocator): SourceL
             throw InvalidProjectDirectory::atPath($installationPath);
         }
 
-        $composerJsonPath  = $realInstallationPath . '/composer.json';
-        $installedJsonPath = $realInstallationPath . '/vendor/composer/installed.json';
+        $composerJsonPath = $realInstallationPath . '/composer.json';
 
         if (! file_exists($composerJsonPath)) {
             throw MissingComposerJson::inProjectPath($installationPath);
         }
 
+        /** @var array{autoload: array{classmap: array<int, string>, files: array<int, string>, psr-4: array<string, array<int, string>>, psr-0: array<string, array<int, string>>}, config: array{vendor-dir: array<string, array<int, string>>}}|null $composer */
+        $composer  = json_decode((string) file_get_contents($composerJsonPath), true);
```

Comment:
BTW, since we now support 7.4+, we can use `JSON_THROW` here

Reply:
I'm not sure I have done this correctly. 
I have added the default `512` as depth.